qiita_db/processing_job.py

```diff
@@ -118,6 +379,80 @@ def exists(cls, job_id):
             qdb.sql_connection.TRN.add(sql, [job_id])
             return qdb.sql_connection.TRN.execute_fetchlast()
 
+    @classmethod
+    def by_ext_id(cls, external_id):
+        """Return Qiita Job UUID associated with external_id
+
+        Parameters
+        ----------
+        external_id : str
+            An external id (e.g. Torque Job ID)
+
+        Returns
+        -------
+        str
+            Qiita Job UUID, if found, otherwise None
+        """
+        with qdb.sql_connection.TRN:
+            sql = """SELECT processing_job_id FROM qiita.processing_job
+                     WHERE external_job_id = '%s'"""
+            qdb.sql_connection.TRN.add(sql, [external_id])
+            return qdb.sql_connection.TRN.execute_fetchlast()
+
+    def get_resource_allocation_info(self):
+        """Return resource allocation defined for this job. For
+        external computational resources only.
+
+        Returns
+        -------
+        str
+            A resource allocation string useful to the external resource
+        """
+        with qdb.sql_connection.TRN:
+            if self.command.name == 'complete_job':
+                type = 'COMPLETE_JOBS_RESOURCE_PARAM'
```

Comment:
Can you rename the `type` variable? It will shadow the `type` built-in function.

Reply:
Thanks for noticing!